src/main/java/com/example/demo/restservice/GreetingController.java

```diff
@@ -1,34 +1,101 @@
 package com.example.demo.restservice;
 
-import java.util.concurrent.atomic.AtomicLong;
-
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.*;
+import java.util.concurrent.atomic.AtomicLong;
+
 
 @RestController
+@RequestMapping("/demo")
 public class GreetingController {
 
     private static final String template = "Hello, %s!";
     private static final String questTemplate = "%s ?";
     private final AtomicLong counter = new AtomicLong();
-
     public Imc test_imc;
+    public Secret secret;
+    public Connection conn;
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @GetMapping("/pushsecret")
+    public Greeting pushSecret(@RequestParam(value = "secret", defaultValue = "mySecret") String pushedSecret,
+                               @RequestParam(value = "key", defaultValue = "mickey") String pushedKey) throws IOException {
+
+        this.createTable();
+
+        this.secret  = new Secret(pushedSecret, pushedKey);
+
+        this.secret.writeSecretToDB(jdbcTemplate);
+        //secret.writeSecretToFile();
+
+        return new Greeting( "Your secret is : " + secret.getSecret(), "Your key for your secret is : " + secret.getKey());
+    }
 
+    @GetMapping("/getsecret")
+    public Greeting getSecret(@RequestParam(value = "key", defaultValue = "") String key) throws IOException, SQLException {
+
+        //get secret from DB
+        String unlockedSecret = getSecretFromDB(key);
+        //get secret from file using key
+        //String unlockedSecret = getSecretFromFile(key);
+
+        return new Greeting("You used this key : " + key, unlockedSecret);
+    }
+
+    public String getSecretFromDB(String key) throws SQLException {
+        String secrets = "";
+        conn = DriverManager.getConnection("jdbc:h2:file:/home/romain/Documents/rest_app/demo/data/secretsDB", "sa", "admin");
+        System.out.println("Creating statement...");
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("SELECT secret FROM Secrets WHERE key = '" + key +"'");
+        while (rs.next()) {
+            secrets = rs.getString(1);
+        }
+        conn.close();
+        String returnedSecret;
+        if (secrets == ""){
+            returnedSecret = "This key doesn't exist.";
+        }
+        else{
+            returnedSecret = "Your secret is : " + secrets;
```

Comment:
As name seems to suggest, `getSecretFromDB` should return a secret (not a sentence)
If you want human readable content you could add logs for exemple here (System.out.prinln("Your secret is" ...))